src/api/app/helpers/webui/request_helper.rb

```diff
@@ -182,4 +182,23 @@ def request_type_of_action(bs_request)
 
     bs_request.bs_request_actions.first.type.titleize
   end
+
+  def build_status_needs_attention?(buildresults)
+    buildresults.any? { |b| ['failed', 'unresolved', 'broken'].include?(b) }
+  end
+
+  def build_status_attention_icon(buildresults)
+    return unless build_status_needs_attention?(buildresults)
+
+    'fa-exclamation-triangle repository-state-broken'
+  end
+
+  def build_status_icon_from_action(action)
+    source_project = Project.find_by_name(action.source_project)
+    source_package = Package.find_by_project_and_name(source_project.name, action.source_package)
```

Comment:
Thanks to @rubhanazeem latest changes in #14244, when there are no source projects, and therefore no build results, the tab does not get rendered and then this code is not called.

Reply:
It's decided by the `BsRequestAction.source_....` which is a string. But if there is a string in there, the project might not exist because it's deleted for instance or because it's read protected etc.